### DISCUSSION TITLE
VariableLinkedBlockingQueue's capacity can be not volatile

### DISCUSSION THREAD

**Discussion by Inuyasha-Monster** [rabbitmq/rabbitmq-java-client#819] [General]
<img width="469" alt="image" src="https://user-images.githubusercontent.com/12888788/181870142-c5ebbe39-ef2d-4c88-8cd7-48228c2da0c4.png">

as title , the field `capacity` not volatile , is there some concurrency issue ?

**Comment by Inuyasha-Monster** [comment]
add another point， that because there is no lock scope in the method that partially accesses the field

**Comment by michaelklishin** [comment]
In the client, this data structure is used to ensure inbound deliveries on a single channel. Since channels must not be shared between threads (this has been documented for many years), I don’t see a reason to use any synchronization mechanism on this field.

In other words, why change it now, a decade into the current consumer dispatch design?

**Reply by Inuyasha-Monster** [reply]
i got it , thx your reply
